Makefile

```diff
@@ -261,34 +261,48 @@ go-tests-short: $(TEST_DEPS) ## Runs comprehensive Go tests with -short flag
 	go test -short -parallel=$$PARALLEL -timeout=$(TEST_TIMEOUT) $(TEST_PKGS)
 .PHONY: go-tests-short
 
-go-tests-short-ci: ## Runs short Go tests with gotestsum for CI (assumes deps built by CI)
+# Internal target for running Go tests with gotestsum for CI
+# Usage: make _go-tests-ci-internal GO_TEST_FLAGS="-short"
+_go-tests-ci-internal:
 	@echo "Setting up test directories..."
 	mkdir -p ./tmp/test-results ./tmp/testlogs
-	@echo 'Running Go tests (short) with gotestsum...'
+	@echo "Running Go tests with gotestsum..."
 	$(DEFAULT_TEST_ENV_VARS) && \
 	$(CI_ENV_VARS) && \
-	gotestsum --format=testname \
-		--junitfile=./tmp/test-results/results.xml \
-		--jsonfile=./tmp/testlogs/log.json \
-		--rerun-fails=3 \
-		--rerun-fails-max-failures=50 \
-		--packages="$(TEST_PKGS) $(RPC_TEST_PKGS) $(FRAUD_PROOF_TEST_PKGS)" \
-		-- -parallel=$$PARALLEL -coverprofile=coverage.out -short -timeout=$(TEST_TIMEOUT) -tags="ci"
+	if [ -n "$$CIRCLE_NODE_TOTAL" ] && [ "$$CIRCLE_NODE_TOTAL" -gt 1 ]; then \
+		export NODE_INDEX=$${CIRCLE_NODE_INDEX:-0} && \
+		export NODE_TOTAL=$${CIRCLE_NODE_TOTAL:-1} && \
+		export PARALLEL_PACKAGES=$$(echo "$(TEST_PKGS) $(RPC_TEST_PKGS) $(FRAUD_PROOF_TEST_PKGS)" | tr ' ' '\n' | awk -v idx=$$NODE_INDEX -v total=$$NODE_TOTAL 'NR % total == idx' | tr '\n' ' ') && \
+		if [ -n "$$PARALLEL_PACKAGES" ]; then \
+			echo "Node $$NODE_INDEX/$$NODE_TOTAL running packages: $$PARALLEL_PACKAGES"; \
+			gotestsum --format=testname \
+				--junitfile=./tmp/test-results/results-$$NODE_INDEX.xml \
+				--jsonfile=./tmp/testlogs/log-$$NODE_INDEX.json \
+				--rerun-fails=3 \
+				--rerun-fails-max-failures=50 \
+				--packages="$$PARALLEL_PACKAGES" \
+				-- -parallel=$$PARALLEL -coverprofile=coverage-$$NODE_INDEX.out $(GO_TEST_FLAGS) -timeout=$(TEST_TIMEOUT) -tags="ci"; \
+		else \
+			echo "Node $$NODE_INDEX/$$NODE_TOTAL has no packages to run"; \
+			touch ./tmp/test-results/results-$$NODE_INDEX.xml; \
+		fi; \
+	else \
+		gotestsum --format=testname \
+			--junitfile=./tmp/test-results/results.xml \
+			--jsonfile=./tmp/testlogs/log.json \
+			--rerun-fails=3 \
+			--rerun-fails-max-failures=50 \
+			--packages="$(TEST_PKGS) $(RPC_TEST_PKGS) $(FRAUD_PROOF_TEST_PKGS)" \
```

Comment:
We probably should de-duplicate the list of packages so that this and the parallel version above don't get out of sync.  Should be able to just assign it to a `ALL_PACKAGES` variable or something.